sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py

```diff
@@ -235,6 +235,9 @@ def receive(self, batch=False, max_batch_size=300, max_wait_time=None):
                     if self._last_received_event:
                         self._offset = self._last_received_event.offset
                     last_exception = self._handle_exception(exception, is_consumer=True)
+                    # If optional dependency is not installed, do not retry.
+                    if isinstance(exception.__cause__, ModuleNotFoundError):
```

Comment:
Is there a reason we wouldn't check for the broader ImportError here (and throughout)?

Reply:
No reason in particular. updated to ImportError everywhere!